mypy/newsemanal/semanal.py

```diff
@@ -3889,53 +3805,86 @@ def current_symbol_kind(self) -> int:
             kind = GDEF
         return kind
 
-    def add_symbol(self, name: str, node: SymbolTableNode,
-                   context: Context) -> None:
+    def add_symbol(self, name: str, node: SymbolNode, context: Context,
+                   module_public: bool = True, module_hidden: bool = False) -> None:
         """Add symbol to the currently active symbol table."""
```

Comment:
Maybe mention in the docstring that all additions to symbol tables should go through this method and adding new ones is discouraged? (Of course, this will not guarantee it will really prevent this, but maybe it will help to at least some extent).